.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,51 @@
+name: Run benchmark for parser
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+permissions:
+  # deployments permission to deploy GitHub pages website
+  deployments: write
+  # contents permission to update benchmark contents in gh-pages branch
+  contents: write
+  # pull-requests permission to create comments on PR in case of alert
+  pull-requests: write
+
+jobs:
+  run-benchmark:
+    name: Performance regression check
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - name: 'Set up JDK'
+      uses: 'actions/setup-java@v4'
+      with:
+        java-version: '11'
+        distribution: 'zulu'
+        cache: 'gradle'
+    - name: Validate Gradle Wrapper
+      uses: gradle/actions/wrapper-validation@v3
+    - name: Setup Gradle
+      uses: gradle/actions/setup-gradle@v3
+      with:
+        gradle-version: wrapper
+    - name: Run benchmark for JVM
+      run: ./gradlew :benchmark:jvmBenchmark
+    - name: Find JVM benchmark result
+      id: results
+      run: echo "report=$(find benchmark/build/reports/benchmarks/main/ -name jvm.json -type f)" >> $GITHUB_OUTPUT
+    - name: Store benchmark result
+      uses: benchmark-action/github-action-benchmark@v1
+      with:
+        name: SnakeKMP benchmark
+        tool: 'jmh'
+        output-file-path: ${{ steps.results.outputs.report }}
+        alert-comment-cc-users: "@krzema12,@aSemy"
```

Comment:
```suggestion
```

The action leaving a comment in the PR is fine, I don't need to be pinged, and I assume `@krzema12` doesn't need to be either.